Modules/posixmodule.c

```diff
@@ -7254,9 +7227,30 @@ os_symlink_impl(PyObject *module, path_t *src, path_t *dst,
 
     Py_BEGIN_ALLOW_THREADS
     /* if src is a directory, ensure target_is_directory==1 */
-    target_is_directory |= _check_dirW(src->wide, dst->wide);
-    result = Py_CreateSymbolicLinkW(dst->wide, src->wide,
-                                    target_is_directory);
+    flags = target_is_directory | _check_dirW(src->wide, dst->wide);
+    if (windows_has_symlink_unprivileged_flag) {
+        /* Allow non-admin symlinks if system allows it: */
+        flags |= SYMBOLIC_LINK_FLAG_ALLOW_UNPRIVILEGED_CREATE;
+    }
+    result = CreateSymbolicLinkW(dst->wide, src->wide, flags);
+    if (windows_has_symlink_unprivileged_flag && !result &&
+        ERROR_INVALID_PARAMETER == GetLastError()) {
+        /* This error might be caused by
+           SYMBOLIC_LINK_FLAG_ALLOW_UNPRIVILEGED_CREATE not being supported.
+           Try again, and update windows_has_symlink_unprivileged_flag if we
+           are successful this time.
+
+           NOTE: There is a risk of a race condition here if there are other
+           conditions than the flag causing ERROR_INVALID_PARAMETER, and
+           another process (or thread) changes that condition in between our
+           calls to CreateSymbolicLink.
+        */
+        flags &= ~(SYMBOLIC_LINK_FLAG_ALLOW_UNPRIVILEGED_CREATE);
+        result = CreateSymbolicLinkW(dst->wide, src->wide, flags);
+        if (result) {
+            windows_has_symlink_unprivileged_flag = FALSE;
+        }
```

Comment:
A note: on systems without the new flag support, we'll do both calls on each `os.symlink` invocation until the first time the symlink can be created. This is probably not important, but could be avoided if `windows_has_symlink_unprivileged_flag` was set to `FALSE` in the case of failure with any error other than `ERROR_INVALID_PARAMETER` (suffering from the same race as the case of success).